core/src/main/java/org/jdbi/v3/core/extension/ExtensionHandler.java

```diff
@@ -110,11 +111,48 @@ static ExtensionHandler createForSpecialMethod(Method method) throws IllegalAcce
      * @return An {@link ExtensionHandler}
      */
     static ExtensionHandler createForMethodHandle(MethodHandle methodHandle) {
-        return (handleSupplier, target, args) -> {
-            if (target == null) {
-                throw new IllegalStateException("no target object present, called from a proxy factory?");
+        return new ExtensionHandler() {
+            @Override
+            public Invoker createInvoker(Object target) {
```

Comment:
here you don't use the qualified name. This make the code harder to read than necessary. I prefer the "ExtensionHandler.Invoker" name everywhere but I would also ask whether it would make sense to simply factor this out into its own class and not use a nested interface.